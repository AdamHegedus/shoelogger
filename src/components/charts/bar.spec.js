import Vue from 'vue';
import Bar from '@/components/charts/bar';

describe('index.vue', () => {

    const shoes = [
        {
            id: 1,
            brand: 'Foo',
            product: 'Shoe A',
            distance: 100
        },
        {
            id: 2,
            brand: 'Foo',
            product: 'Shoe B',
            distance: 50
        },
        {
            id: 3,
            brand: 'Bar',
            product: 'Shoe C',
            distance: 0
        },
        {
            id: 4,
            brand: 'Bar',
            product: 'Shoe D',
            distance: 15
        }
    ];

    const Constructor = Vue.extend(Bar);
    const vm = new Constructor({
        propsData: {
            data: shoes
        }
    }).$mount(document.createElement('div'));

    it('should match snapshot', () => {
        // ASSIGN

        // ACT
        const actual = vm.$el;

        // ASSERT
        expect(actual).toMatchSnapshot();
    });
});
