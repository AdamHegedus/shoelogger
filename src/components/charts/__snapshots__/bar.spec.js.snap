// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue should match snapshot when data is available 1`] = `
<div
  class="bar-chart"
>
  <svg
    height="200"
    width="200"
  >
    <g
      transform="translate(20, 20)"
    >
      <g
        class="axis axis-x"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0, 130)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H160.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(32.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(64.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            40
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(96.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(128.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            80
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(160.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            100
          </text>
        </g>
        <text
          class="axis-label"
          x="160"
          y="25"
        >
          Distance (km)
        </text>
      </g>
      <g
        class="bars"
      >
        <rect
          class="bar product-shoe-a"
          height="16"
          width="160"
          x="0"
          y="105"
        />
        <rect
          class="bar product-shoe-b"
          height="16"
          width="80"
          x="0"
          y="73"
        />
        <rect
          class="bar product-shoe-c"
          height="16"
          width="0"
          x="0"
          y="41"
        />
        <rect
          class="bar product-shoe-d"
          height="16"
          width="24"
          x="0"
          y="9"
        />
      </g>
      <g
        class="data-labels"
      >
        <text
          class="product-label"
          dy="0.5em"
          x="0"
          y="97"
        >
          Shoe A
        </text>
        <text
          class="product-label"
          dy="0.5em"
          x="0"
          y="65"
        >
          Shoe B
        </text>
        <text
          class="product-label"
          dy="0.5em"
          x="0"
          y="33"
        >
          Shoe C
        </text>
        <text
          class="product-label"
          dy="0.5em"
          x="0"
          y="1"
        >
          Shoe D
        </text>
      </g>
      <g
        class="data-labels"
      >
        <text
          class="data-label"
          dy="0.5em"
          x="160"
          y="97"
        >
          100 km
        </text>
        <text
          class="data-label"
          dy="0.5em"
          x="160"
          y="65"
        >
          50 km
        </text>
        <text
          class="data-label"
          dy="0.5em"
          x="160"
          y="33"
        >
          0 km
        </text>
        <text
          class="data-label"
          dy="0.5em"
          x="160"
          y="1"
        >
          15 km
        </text>
      </g>
    </g>
  </svg>
</div>
`;

exports[`index.vue should match snapshot when data is empty 1`] = `
<div
  class="bar-chart"
>
  <svg
    height="200"
    width="200"
  >
    <g
      transform="translate(20, 20)"
    >
      <g
        class="axis axis-x"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0, 130)"
      >
        <path
          class="domain"
          d="M0.5,6V0.5H160.5V6"
          stroke="currentColor"
        />
        <text
          class="axis-label"
          x="160"
          y="25"
        >
          Distance (km)
        </text>
      </g>
      <g
        class="bars"
      />
      <g
        class="data-labels"
      />
      <g
        class="data-labels"
      />
    </g>
  </svg>
</div>
`;
